### PR TITLE
Dashboard: Align module Protect data logic with Akismet

### PR DESCRIPTION
**Pull request by dognose24**
## Proposed changes:

Since we are tackling the Jetpack Stats Widget that consumes the API response of modules data, aligning module Protect logic with Akismet for the status check would improve the consistency.

* Align the consumed data from wp option `jetpack_protect_blocked_attempts` with the format of `akismet_spam_count`.
* Use only numeric and plugin status data for Akismet UI condition check.
* Align module Protect UI condition check logic with Akismet.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

1. Start on a test site with Brute force protection and Akismet Anti-spam enabled. The loading status of Brute force protection should be aligned with Akismet.

截圖 2023-04-21 上午12 54 11

2. Update the wp options of `jetpack_protect_blocked_attempts` and `akismet_spam_count` to `0`, and the status text of these two cards should be aligned.

截圖 2023-04-21 上午1 04 07

3. Delete the wp options of `jetpack_protect_blocked_attempts` and `akismet_spam_count`, and the status text should be the same as `0`.

4. Set the wp options of `jetpack_protect_blocked_attempts` and `akismet_spam_count` to `1`, and the counts of two cards should be aligned.

截圖 2023-04-21 上午12 54 44

### CODE REVIEW

**Comment by kangzj** [comment]
Good effort improving this @dognose24 !

<strike>I'm just wondering, would it be better we show the number even if they are zero? which give user an expression that it's started working. What do you think?</strike>

Let's respect the old design atm 😄

**Review by kangzj — DISMISSED** [review]
LGTM

File: tools/eslint-excludelist.json [hunk 1]
```diff
@@ -31,7 +31,6 @@
 	"projects/plugins/jetpack/_inc/client/notices/index.js",
 	"projects/plugins/jetpack/_inc/client/pro-status/index.jsx",
 	"projects/plugins/jetpack/_inc/client/security/sso.jsx",
-	"projects/plugins/jetpack/_inc/client/state/at-a-glance/reducer.js",
```
Comment (kangzj) [inline tools/eslint-excludelist.json:32]: 👍

Comment (dognose24) [reply tools/eslint-excludelist.json:32]: 👍🏼

Comment (bluefuton) [reply tools/eslint-excludelist.json:32]: @dognose24 I noticed this was preventing the build from passing, so the bot kept reapplying the 'Needs author reply' label. This seemed to sort it out!

**Comment by bluefuton** [comment]
Thank you for taking the time to fix this up @dognose24! 🙌

File: tools/eslint-excludelist.json (hunk 1 above)
Comment (dognose24) [reply tools/eslint-excludelist.json:32]: @bluefuton Thanks for the explanation! 🙇🏼‍♂️

**Comment by dognose24** [comment]
@jeherve, if you have time, I would appreciate your reviewing this PR to keep the Akismet logic consistent since you are the author of [recent changes](https://github.com/Automattic/jetpack/pull/29866). 🙂

**Review by zinigor — APPROVED** [review]

**Comment by dognose24** [comment]
@zinigor Thanks for your review!

**Pull request merged**